examples/mean-reversion.go

```diff
@@ -0,0 +1,255 @@
+	position, err := alp.client.GetPosition(alpacaClient.stock)
+	if err != nil {
+	} else {
+		positionQty = int(position.Qty.IntPart())
+		positionVal, _ = position.MarketValue.Float64()
+	}
+
+	// Get the new updated price and running average.
+	bars, _ := alp.client.GetSymbolBars(alpacaClient.stock, alpaca.ListBarParams{Timeframe: "minute"})
+	currPrice := float64(bars[len(bars)-1].Close)
+	alpacaClient.closingPrices = append(alpacaClient.closingPrices, currPrice)
+	if len(alpacaClient.closingPrices) > 20 {
+		alpacaClient.closingPrices = alpacaClient.closingPrices[1:]
+	}
+	alpacaClient.runningAverage = ((alpacaClient.runningAverage * float64(len(alpacaClient.closingPrices)-1)) + currPrice) / float64(len(alpacaClient.closingPrices))
+
+	if currPrice > alpacaClient.runningAverage {
+		// Sell our position if the price is above the running average, if any.
+		if positionQty > 0 {
+			fmt.Println("Setting position to zero")
+			alp.submitLimitOrder(positionQty, alpacaClient.stock, currPrice, "sell")
+		} else {
+			fmt.Println("No position in the stock.  No action required.")
+		}
+	} else if currPrice < alpacaClient.runningAverage {
+		// Determine optimal amount of shares based on portfolio and market data.
+		account, _ := alp.client.GetAccount()
+		buyingPower, _ := account.BuyingPower.Float64()
+		positions, _ := alp.client.ListPositions()
+		portfolioVal, _ := account.Cash.Float64()
+		for _, position := range positions {
+			rawVal, _ := position.MarketValue.Float64()
+			portfolioVal += rawVal
+		}
+		portfolioShare := (alpacaClient.runningAverage - currPrice) / currPrice * 200
+		targetPositionValue := portfolioVal * portfolioShare
+		amountToAdd := targetPositionValue - positionVal
+
+		// Add to our position, constrained by our buying power; or, sell down to optimal amount of shares.
+		if amountToAdd > 0 {
+			if amountToAdd > buyingPower {
+				amountToAdd = buyingPower
+			}
+			var qtyToBuy = int(amountToAdd / currPrice)
+			alp.submitLimitOrder(qtyToBuy, alpacaClient.stock, currPrice, "buy")
+		} else {
+			amountToAdd *= -1
+			var qtyToSell = int(amountToAdd / currPrice)
+			if qtyToSell > positionQty {
+				qtyToSell = positionQty
+			}
+			alp.submitLimitOrder(qtyToSell, alpacaClient.stock, currPrice, "buy")
+		}
+	}
+}
+
+// Submit a limit order if quantity is above 0.
+func (alp alpacaClientContainer) submitLimitOrder(qty int, symbol string, price float64, side string) error {
+	account, _ := alp.client.GetAccount()
+	if qty > 0 {
+		adjSide := alpaca.Side(side)
+		limPrice := decimal.NewFromFloat(price)
+		order, err := alp.client.PlaceOrder(alpaca.PlaceOrderRequest{
+			AccountID:   account.ID,
+			AssetKey:    &symbol,
+			Qty:         decimal.NewFromFloat(float64(qty)),
+			Side:        adjSide,
+			Type:        "limit",
+			LimitPrice:  &limPrice,
+			TimeInForce: "day",
+		})
+		if err == nil {
+			fmt.Println("Limit order of " + "|" + strconv.Itoa(qty) + " " + symbol + " " + side + "|" + " completed.")
```

Comment:
Would rather this be `fmt.Printf("Limit order of | %d %s %s | completed", qty, symbol, side)` or so.